inc/azure_macro_utils/macro_utils.h

```diff
@@ -12696,6 +12696,1728 @@ MU_IF(X, "true", "false") => "true"
 #define MU_IF1(trueBranch, falseBranch) trueBranch
 
 
+/*the following macro want to eat empty arguments from a list */
+/*examples:                                                   */
+/*MU_EAT_EMPTY_ARGS(, , X) expands to X                       */
+
+#define MU_EAT_EMPTY_ARGS0() 
+#define MU_EAT_EMPTY_ARGS1(x) x
```

Comment:
Seems a little bit much to generate each one. I would highly consider using the existing infra for looping (FOR_EACH).

I tested this and it passed the tests:

#define MU_EXPAND_TO_NOTHING(arg)
#define MU_EAT_EMPTY_ARG(arg_count, arg) MU_IF(MU_ISEMPTY(arg),,arg) MU_IF(MU_ISEMPTY(arg),MU_EXPAND_TO_NOTHING,MU_IFCOMMALOGIC)(MU_DEC(arg_count))
#define MU_EAT_EMPTY_ARGS(...) MU_FOR_EACH_1_COUNTED(MU_EAT_EMPTY_ARG, __VA_ARGS__)

Reply:
I bow before the master.

---
In reply to: [270708972](https://github.com/Azure/azure-macro-utils-c/pull/11#discussion_r270708972) [](ancestors = 270708972)